core/src/raw/oio/write/append_write.rs

```diff
@@ -96,8 +96,8 @@ where
         Ok(())
     }
 
-    async fn close(&mut self) -> Result<()> {
-        Ok(())
+    async fn close(&mut self) -> Result<Metadata> {
+        Ok(Metadata::default())
```

Comment:
Perhaps we should return valid data here. The metadata could be returned and replaced by `self.inner.append().await`, then stored within `AppendWriter` until the user calls `close`.